### ISSUE TITLE
Generated components have TypeScript errors

### ISSUE THREAD

**Issue by benoit-ctrl** [gluestack/gluestack-ui#2518]
Firstly, thanks for you work on Gluestack and for the community !
I encountered a TypeScript problem when using some Gluestack components so I decided to share it here.

### Description

In some components like Heading and Icon there is TypeScript errors

### CodeSandbox/Snack link

(none, failed to install Tailwind in Snack)

### Steps to reproduce

1. Install Heading or Icon component for example
2. Check the generated files in your IDE with integrated TypeScript service or try to compile manually with `tsc` 
3. Observe reported TypeScript errors

### gluestack-ui Version

2 (latest)

### Platform

- [X] Expo
- [ ] React Native CLI
- [ ] Next
- [ ] Web
- [ ] Android
- [ ] iOS

### Additional Information

- TypeScript `5.6.2`
- React `18.2.0`

<img width="812" alt="image" src="https://github.com/user-attachments/assets/91d30fc9-c269-4452-b55e-cc4830713f00">
<img width="786" alt="image" src="https://github.com/user-attachments/assets/128d7b84-6ad4-471c-80bb-707003b4dc6c">

**Comment by Viraj-10** [comment]
Hey @benoit-ctrl, This issues will be fixed in #2475.

**Comment by benoit-ctrl** [comment]
Hello @Viraj-10, thank you for your answer👌 !

I have a question, since Gluestack is a "copy-paste components" library the only way to get update is to re-generate components after a library update ? 
If so, then if we've already customized Gluestack components it's starts to be more complicated I suppose

**Comment by Viraj-10** [comment]
@benoit-ctrl, Yes, You should just create a branch from you main and add all the installed components.
you will be able to see all the changes through diff. discard all the unwanted changes and commit your changes. 

We know this is a pain point for all the customised components. we are planning to split style files from components. so, it would be easier to update. 

Also we will be publishing Diff with every release and releases would be on weekly bases and planning on adding `update` command.

**Comment by benoit-ctrl** [comment]
@Viraj-10 Thank you for your detailed answer, I understand better now 😁

It's a great idea trying to support a little the updating process through tools, if it doesn't have too much impact on the other milestones of the project
